koku/api/report/queries.py

```diff
@@ -683,11 +706,15 @@ def _get_tag_group_by(self):
         tag_groups = self.get_tag_group_by_keys()
         for tag in tag_groups:
             original_tag = strip_prefix(tag, TAG_PREFIX)
-            sanitized_tag = sanitize_tag(original_tag)
-            tag_db_name = self._mapper.tag_column + "__" + sanitized_tag
             encoded_tag_url = quote(original_tag, safe=URL_ENCODED_SAFE)
             group_pos = self.parameters.url_data.index(encoded_tag_url)
-            group_by.append((tag_db_name, group_pos))
+            safe_tag = safe_column_alias(original_tag)
+            tag_db_name = self._mapper.tag_column + "__" + safe_tag
```

Comment:
This is fundamentally broken. By changing the tag value, you've changed what we are querying in the database. It is impossible for this work because we need this value as-is to do a group-by. If the original value for a key in the database was "key with space", and we try to group on that, what we end up grouping by in the db is "key_with_space". This is wrong and breaks the query.

We need a solution that passes the original value to the query, else it is wrong.